core/src/Kokkos_Macros.hpp

```diff
@@ -521,6 +521,7 @@
 #if defined ( KOKKOS_ENABLE_CUDA )
   #if ( 9000 <= CUDA_VERSION )
   #define KOKKOS_IMPL_CUDA_VERSION_9_WORKAROUND
+  #define KOKKOS_IMPL_CUDA_SYNCWARP_NEEDS_MASK
```

Comment:
This should only be defined if `__CUDA_ARCH__` is defined as well. That may remove the need for the dummy function altogether.

Reply:
I'll add that and check, thanks!

Reply:
adding a `__CUDA_ARCH__` check ended up breaking the build, I need to look at this a bit more carefully...